docs/src/puppeteer-js.md

```diff
@@ -0,0 +1,110 @@
+---
+id: puppeteer
+title: "Migrating from Puppeteer"
+---
+
+<!-- TOC -->
+
+## Migration Principles
+
+This guide describes migration to [Playwright Library](./library) from Puppeteer. The APIs have similarities, but Playwright offers much more possibilities for web testing and cross-browser automation.
+
+- Most Puppeteer APIs can be used as is
+- Playwright is cross-browser
+- You probably don't need explicit wait
+
+## Cheat Sheet
+
+| Puppeteer                                          | Playwright Library                          |
+|----------------------------------------------------|---------------------------------------------|
+| `await puppeteer.launch(...)`                      | `await playwright.chromium.launch(...)`     |
+| `await browser.createIncognitoBrowserContext(...)` | `await browser.newContext(...)`             |
+| `await page.setViewport(...)`                      | `await page.setViewportSize(...)`           |
+| `await page.waitForSelector(selector)`<br/>`await page.click(selector);` | `await page.click(selector)` |
+| `await page.waitForXPath(XPathSelector)`           | `await page.waitForSelector(XPathSelector)` |
+| `await page.$(...)`                                | Discouraged, use [Locators](./api/class-locator) instead |
+| `await page.$x(xpath_selector)`                    | `await page.$(xpath_selector)`              |
+| `await page.waitForNetworkIdle(...)`               | `await page.waitForLoadState({ state: 'networkidle' }})` |
+| `await page.cookies([...urls])`                    | `await browserContext.cookies([urls])`      |
+| `await page.deleteCookie(...cookies)`              | `await browserContext.clearCookies()`       |
+| `await page.setCookie(...cookies)`                 | `await browserContext.addCookies(cookies)`  |
+| `page.on('request', ...)`                          | Handled through [`method: Page.route`]. |
+| `await page.waitForFileChooser(...)`<br/>`await elementHandle.uploadFile(...)`              | `await elementHandle.setInputFiles(...)`    |
+
+`page.waitForNavigation` and `page.waitForSelector` remain, but in many cases will not be necessary due to [auto-waiting](./actionability).
+
+## Example
```

Comment:
Adding a test comparison will highlight the improvement that `@playwright/test` adds

Reply:
My intent here was to offer a simple automation example. Indeed it can be useful to demonstrate `@playwright/test`

Do you think I should add a sample in this "Example" section, or in the "Testing" one ?

Reply:
I think we can have just a example using a test, similar to https://playwright.dev/docs/protractor#example or maybe keep the current script as well, I'm not sure